### PR TITLE
Including the possibility to automatically remove data from collections using TTL or collection size

### PR DESCRIPTION
**Pull request by gtorodelvalle**
- Fixes https://github.com/telefonicaid/IoT-STH/issues/147
- 100% tests passed
- Assigned to @frbattid 

This PR should be landed once https://github.com/telefonicaid/IoT-STH/pull/146 lands in `develop`.

### CODE REVIEW

File: src/sth_database.js [hunk 1]
```diff
@@ -300,9 +306,11 @@
    *  if it does not exist
    * @param {boolean} shouldStoreHash Flag indicating if the collection hash should be stored in case the collection
    *  is created
+   * @param {boolean} shouldTruncate Flag indicating if the collection should be truncate according to the requested data
```
Comment (frbattid) [inline src/sth_database.js]: "...if the collection should be truncate..." --> "...if the collection should be truncated..."

File: src/sth_database.js [hunk 2]
```diff
@@ -10,8 +10,14 @@
 
   var sthConfig, sthLogger, sthHelper, connectionURL, eventSchema, aggregatedSchema;
 
-  var MAX_NAMESPACE_SIZE_IN_BYTES = 120,
-      MIN_HASH_SIZE_IN_BYTES = 20;
+  /**
+   * The maximumum namespace size in bytes has to be decreased to 113 bytes
```
Comment (frbattid) [inline src/sth_database.js]: I need more detail o this :)

Comment (gtorodelvalle) [inline src/sth_database.js]: :D Probably me too... :p The point is that to apply the TTL using Mongo, a new index has to be created on the corresponding temporal property (see http://docs.mongodb.org/v2.6/tutorial/expire-data/ ). That index is created in the default `system.indexes` collection and it seems it is not possible to create that index for aggregated data collections if the hashing mechanism is activated and more than 113 bytes are used for the whole namespace (database + collection name) :) I cannot say much more than that... :D

File: src/sth_database.js (hunk 1 above)
Comment (gtorodelvalle) [inline src/sth_database.js]: Fixed in https://github.com/telefonicaid/IoT-STH/commit/b36a70fdb47783cd5022406b65d6dcd1c1a3a60a

File: src/sth_database.js (hunk 2 above)
Comment (frbattid) [inline src/sth_database.js]: OK, that's enough for me :) 113 bytes... and decrasing! :p

File: src/sth_database.js [hunk 3]
```diff
@@ -330,38 +338,107 @@
...
+    };
+
+    connection.db.collection(collectionName, {strict: true},
+      function (err, collection) {
+        if (err &&
+          (err.message === 'Collection ' + collectionName + ' does not exist. Currently in strict mode.') &&
+            shouldCreate) {
+          if (shouldTruncate && !isAggregated) {
+            // Set the size removal policy if required
+            if (parseInt(sthConfig.TRUNCATION_SIZE) > 0) {
```
Comment (frbattid) [inline src/sth_database.js]: A log trace could be created when the trucntaction size is not > 0.

Comment (frbattid) [inline src/sth_database.js]: I'm wondering if some kind of input parameters validation is done. If that's the case, then is not necessary to ckeck for it here.

Comment (gtorodelvalle) [inline src/sth_database.js]: In fact, the validation done on the configuration parameters is very "light", so I suggest creating a new issue on the matter so we can solve it properly for all the configuration parameters. I created https://github.com/telefonicaid/IoT-STH/issues/161 for this. Are you OK with this?

Comment (frbattid) [inline src/sth_database.js]: Fine for me :)

**Comment by frbattid** [comment]
LGTM

**Pull request merged**